### PR TITLE
Add ff docs for secondaryKeyAsGSI

### PR DESCRIPTION
**Pull request by SwaySway**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: client/src/amplify-ui/cli-feature-flag/feature-flags.json [hunk 1]
```diff
@@ -84,6 +84,27 @@
             "defaultExistingProject": true
           }
         ]
+      },
+      "secondaryKeyAsGSI": {
+        "description": "Changes the behaviour of @key to remove the creation of an LSI.",
```
Comment (edwardfoyle) [inline client/src/amplify-ui/cli-feature-flag/feature-flags.json:89]: I'd suggest: "Changes the behavior of @key to always create GSIs"

Comment (renebrandel) [reply client/src/amplify-ui/cli-feature-flag/feature-flags.json:89]: Changes the @key behavior to always create GSIs.

File: client/src/amplify-ui/cli-feature-flag/feature-flags.json [hunk 2]
```diff
@@ -84,6 +84,27 @@
...
+        "type": "Feature",
+        "valueType": "Boolean",
+        "versionAdded": "4.41.x",
+        "deprecationDate": "May 1st 2021",
+        "values": [
+          {
+            "value": "true",
+            "description": "Will default to creating a GSI when using @key",
```
Comment (renebrandel) [inline client/src/amplify-ui/cli-feature-flag/feature-flags.json:97]: @key directive always creates GSIs

File: client/src/amplify-ui/cli-feature-flag/feature-flags.json [hunk 3]
```diff
@@ -84,6 +84,27 @@
...
+            "defaultNewProject": true,
+            "defaultExistingProject": false
+          },
+          {
+            "value": "false",
+            "description": "Will use old behavior to create a LSI when the using a primary field.",
```
Comment (renebrandel) [inline client/src/amplify-ui/cli-feature-flag/feature-flags.json:103]: @key directive creates LSI when using a primary field.

File: client/src/amplify-ui/cli-feature-flag/feature-flags.json (hunk 2 above)
Comment (edwardfoyle) [inline client/src/amplify-ui/cli-feature-flag/feature-flags.json:97]: Will create a GSI when using a named @key

File: client/src/amplify-ui/cli-feature-flag/feature-flags.json (hunk 3 above)
Comment (edwardfoyle) [inline client/src/amplify-ui/cli-feature-flag/feature-flags.json:103]: Will use old behavior of creating an LSI when an @key directive uses a primary key field.

**Review by edwardfoyle — APPROVED** [review]

**Review by kaustavghosh06 — APPROVED** [review]

**Pull request merged**